tablenamer/tablenamer.go

```diff
@@ -0,0 +1,115 @@
+package tablenamer
+
+import (
+	"unicode"
+
+	"github.com/samonzeweb/godb/tablenamer/plural"
+	"sync"
+)
+
+// TableNamer function type is used set table naming function to build table name from struct's name
+type TableNamerFn func(string, bool) string
+
+// snakeCache is cacher for ToSnakeCase function
+type snakeCache struct {
```

Comment:
This cache will have very few writes and many reads. For avoid performance issues due to lock it's better to use https://golang.org/pkg/sync/#Map
It will probably break compatibility with Go 1.8 , but its support will end in two months.